numba/tests/test_stencils.py

```diff
@@ -365,12 +407,1272 @@ def test_impl(n):
+
+    # This fails pyStencil but stencil func runs, fairly sure it shouldn't?
+    def test_basic42(self):
+        """2 args! rel arrays very close in size"""
+        def kernel(a, b):
+            return a[0, 1] + b[0, -2]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(9.).reshape(3, 3)
+        self.check(kernel, a, b, expected_exception=[ValueError, LoweringError])
+
+    def test_basic43(self):
+        """2 args more complexity"""
+        def kernel(a, b):
+            return a[0, 1] + a[1, 2] + b[-2, 0] + b[0, -1]
+        a = np.arange(30.).reshape(5, 6)
+        b = np.arange(30.).reshape(5, 6)
+        self.check(kernel, a, b)
+
+    # doesn't match pyStencil, probably expected
+    def test_basic44(self):
+        """2 args, has assignment before use"""
+        def kernel(a, b):
+            a[0, 1] = 12
+            return a[0, 1]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, expected_exception=[ValueError, LoweringError])
+
+    # doesn't match pyStencil, probably expected
+    def test_basic45(self):
+        """2 args, has assignment and then cross dependency"""
+        def kernel(a, b):
+            a[0, 1] = 12
+            return a[0, 1] + a[1, 0]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, expected_exception=[ValueError, LoweringError])
+
+    # doesn't match pyStencil, probably expected
+    def test_basic46(self):
+        """2 args, has cross relidx assignment"""
+        def kernel(a, b):
+            a[0, 1] = b[1, 2]
+            return a[0, 1] + a[1, 0]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, expected_exception=[ValueError, LoweringError])
+
+    def test_basic47(self):
+        """3 args"""
+        def kernel(a, b, c):
+            return a[0, 1] + b[1, 0] + c[-1, 0]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        c = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, c)
+
+    # matches pyStencil, but all ought to fail
+    # probably hard to detect?
+    def test_basic48(self):
+        """2 args, has assignment before use via memory alias"""
+        def kernel(a):
+            c = a.T
+            c[:, :] = 10
+            return a[0, 1]
+        a = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a)
+
+
+    def test_basic49(self):
+        """2 args, standard_indexing on second"""
+        def kernel(a, b):
+            return a[0, 1] + b[0, 3]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'})
+
+    # this should raise as b[0, 15] is out of bounds for 'b' under standard_indexing
+    def test_basic50(self):
+        """2 args, standard_indexing OOB"""
+        def kernel(a, b):
```

Comment:
Think this needs fixing as it is OOB access in `b`?

Reply:
The following doesn't assert either.  So, we are consistent.  A static check might catch this case but in general would be useless.  A dynamic check I guess is considered too expensive so that is why you don't do it either.

@njit
def f1(b):
    return b[0,15]

b = np.arange(12.).reshape(3,4)

res = f1(b)
print("res", res)

Reply:
Yes, I was wondering with the additional analysis/code gen if this is cheap to do? but as noted it'd probably be too expensive at run time. Let's leave this one, behaviour is consistent with existing.